tests/conftest.py

```diff
@@ -1 +1 @@
-pytest_plugins = ['pytest_jupyter_server']
\ No newline at end of file
+pytest_plugins = ['jupyter_server.pytest_plugin']
```

Comment:
I guess I'm confused on when this is necessary.  Given these tests will depend on pytest-jupyter's `pytest_jupytercore` and `pytest_jupyterserver` plugins, do consumers of those fixtures (including jupyter_server) require a `pytest_plugins` statement or is the extra-requires [test] dependency on `pytest-jupyter` sufficient?

In working with #335, it doesn't seem like this entry (referencing the now external plugins) makes any difference.

Reply:
If the `pytest11` entry point in this repo is removed, there would be no particular way for pytest to know that it should use it, it wouldn't be imported, and the issue wouldn't occur for folk that have packages installed that try to actually `.load` their entrypoints.

Of the [seven ways](https://docs.pytest.org/en/stable/writing_plugins.html#plugin-discovery-order-at-tool-startup) a plugin can be found, if it a test dependency doesn't use number 4, i'm a big fan of being explicit in `conftest.py` (number 7) or command line  (number 1), but checked in as checked in inside a file that ends up in the distribution, e.g. `pyproject.toml`, or `setup.cfg` rather than rely on env vars, switches, etc. 

> do consumers of those fixtures (including jupyter_server) require a pytest_plugins statement 

Once it's an explicit, real dependency, on a dedicated, test-time package, then everything is cool, and no `pytest_plugin` variable will be required, as number 4 will take care of it.

So generally: you'd never want anything referenced by Other People's Entry Points (what you might not control) to be behind an `extra`... depending on how they are used (e.g. have to be actually imported).

The whole prefix thing from #335? Yeah, well, that's just what a test regume is buying into with magic named functions. I'm not a huge fan, as the pattern (which, luckily, seems to be rather unique) doesn't play nice-nice with other tools (e.g. mypy, etc). But it's the only game in town, aside from the few little ecosystems where they felt they had to home-roll their own thing... `noflo`, `robotframework`, etc. have extremely unpythonic things they do, in the name of specific purposes.